cupy/cuda/cub.pyx

```diff
@@ -202,9 +210,10 @@ def device_segmented_reduce(ndarray x, op, axis, out=None,
         op, dtype_id)
     ws = memory.alloc(ws_size)
     ws_ptr = <void*>ws.ptr
-    cub_device_segmented_reduce(ws_ptr, ws_size, x_ptr, y_ptr, n_segments,
-                                offset_start_ptr, offset_end_ptr, s,
-                                op, dtype_id)
+    with nogil:
+        cub_device_segmented_reduce(ws_ptr, ws_size, x_ptr, y_ptr, n_segments,
```

Comment:
Ditto. How about removing `with nogil:`?

Reply:
`cub_device_segmented_reduce` is actually batched reduce method so I want to add `nogil`.